clang/test/SemaSYCL/attr-add-ir-attributes-function-conflict.cpp

```diff
@@ -5,19 +5,48 @@
 
 #include "sycl.hpp"
 
-constexpr const char AttrName1[] = "Attr1";
-constexpr const char AttrVal1[] = "Val1";
+struct NameValuePair {
+  static constexpr const char *name = "Attr1";
+  static constexpr const int value = 1;
+};
+
+template <typename... Pairs> struct Wrapper {
+  template <typename KernelName, typename KernelType>
+  [[__sycl_detail__::add_ir_attributes_function(Pairs::name..., Pairs::value...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+    kernelFunc();
+  }
+};
+
+template <typename... Pairs> struct WrapperWithImplicit {
+  template <typename KernelName, typename KernelType>
+  [[__sycl_detail__::add_ir_attributes_function("sycl-single-task", Pairs::name..., 0, Pairs::value...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+    kernelFunc();
+  }
+};
+
+template <typename... Pairs> struct WrapperWithFilter {
+  template <typename KernelName, typename KernelType>
+  [[__sycl_detail__::add_ir_attributes_function({"some-filter-string"}, Pairs::name..., Pairs::value...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+    kernelFunc();
+  }
+};
 
-template <const char *... Strs> struct Wrapper {
+template <typename... Pairs> struct WrapperWithImplicitAndFilter {
   template <typename KernelName, typename KernelType>
-  [[__sycl_detail__::add_ir_attributes_function(Strs...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
+  [[__sycl_detail__::add_ir_attributes_function({"some-filter-string"}, "sycl-single-task", Pairs::name..., 0, Pairs::value...)]] __attribute__((sycl_kernel)) void kernel_single_task(const KernelType &kernelFunc) {
     kernelFunc();
   }
 };
 
 int main() {
   Wrapper<> EmptyWrapper;
-  Wrapper<AttrName1, AttrVal1> NonemptyWrapper;
+  Wrapper<NameValuePair> NonemptyWrapper;
+  WrapperWithImplicit<> EmptyWrapperWithImplicit;
+  WrapperWithImplicit<NameValuePair> NonemptyWrapperWithImplicit;
+  Wrapper<> EmptyWrapperWithFilter;
```

Comment:
Shouldn't this be WrapperWithFilter<> EmptyWrapperWithFilter;

Reply:
Good eye! It has been fixed. Also showed an issue in `CheckSYCLAddIRAttributesFunctionAttrConflicts` which has  been fixed too.